### ISSUE TITLE
create visualization plot for parameter optimization

### ISSUE THREAD

**Issue by rythorpe** [jonescompneurolab/hnn-core#423]

**Comment by jasmainak** [comment]
@rythorpe this came up today. One of the issues with the optimization function is that it does not update `net._params`. I see users trying to access parameters through that. Is this attribute still needed?

Perhaps more urgently, we should have a simple function that extracts the optimized parameters from a network object and spits them out as a dictionary? So users can report in their publications what is the new parameters. This use case came up in a discussion today.

**Comment by rythorpe** [comment]
Yeah, that attribute is private for a reason ;-) Optimization doesn't touch that attribute, but rather modifies the specific drive attributes of the `Network` instance.

[_get_drive_params()](https://github.com/jonescompneurolab/hnn-core/blob/3dce2f1dbdc9e7097ba9c3677da385f20d95edff/hnn_core/optimization.py#L368) was designed with the intention of eventually becoming public so that users could extract the relevant optimized parameters.

**Comment by jasmainak** [comment]
I think the users ideally want just a flat dictionary ... is that returned by this function?

**Comment by rythorpe** [comment]
Nope, but the bones are there. We'll need to be thoughtful here about not recreating the whole .param file debacle as we want to provide user-facing parameters that have meaning + context (i.e., as opposed to black-box parameters).

**Comment by ntolley** [comment]
@samadpls this could be something to consider for your project!
